lib/Jsrt/ChakraCommon.h

```diff
@@ -23,8 +23,49 @@
 #ifndef _CHAKRACOMMON_H_
 #define _CHAKRACOMMON_H_
 
+// Platform specific code
+#if defined(_WIN32) && defined(_MSC_VER)
 #include <oaidl.h>
 
+// Header macros
+#define CHAKRA_CALLBACK CALLBACK
+#define CHAKRA_API STDAPI_(JsErrorCode)
+
+typedef DWORD_PTR ChakraCookie;
+typedef BYTE* ChakraBytePtr;
+#else
+#include <cstdint>
+
+// SAL compat
+#define _Return_type_success_(x)
+#define _In_
+#define _In_z_
+#define _In_opt_
+#define _Inout_
+#define _Out_
+#define _Out_opt_
+#define _In_reads_(x)
+#define _Pre_maybenull_
+#define _Pre_writable_byte_size_(byteLength)
+#define _Outptr_result_buffer_(byteLength)
+#define _Outptr_result_bytebuffer_(byteLength)
+#define _Outptr_result_z_
+#define _Ret_maybenull_
+#define _Out_writes_to_opt_(byteLength, byteLength2)
+
+// Header macros
+// REVIEW: What should we do here??
+#ifdef __i386___
+#define CHAKRA_CALLBACK __attribute__((cdecl))
+#else
+#define CHAKRA_CALLBACK
+#endif
+
+#define CHAKRA_API extern "C" JsErrorCode
+
+typedef uintptr_t ChakraCookie;
+typedef unsigned char* ChakraBytePtr;
+#endif
```

Comment:
Can we add comments at all #else and #endif to mark which #if it refers to? makes it easier to read the definition blocks